dvc/render/converter/vega.py

```diff
@@ -0,0 +1,316 @@
+                res["field"] = field
+            return res
+
+        def _generate_y(datapoint, field):
+            tmp = datapoint.copy()
+            tmp["dvc_inferred_y_value"] = datapoint[field]
+            tmp = _add_version_info(
+                tmp, _version_info(revision, filename, field)
+            )
+            if (
+                "y_label" not in properties_update
+                and "y_label" not in self.plot_properties
+            ):
+                properties_update["y_label"] = "y"
+
+            properties_update["y"] = "dvc_inferred_y_value"
+
+            return tmp
+
+        if not y_values:
+            for dp in datapoints:
+                result.append(
+                    _add_version_info(dp, _version_info(revision, filename))
+                )
+
+        if isinstance(y_values, str):
+            for datapoint in datapoints:
+                result.append(
+                    _add_version_info(
+                        datapoint, _version_info(revision, filename, y_values)
+                    )
+                )
+
+        if isinstance(y_values, list):
+            for datapoint in datapoints:
+                for y_val in y_values:
+                    if y_val in datapoint:
+                        result.append(_generate_y(datapoint, y_val))
+
+        if isinstance(y_values, dict):
+
+            def _to_set(values: Iterable):
+                result = set()
+                for val in values:
+                    if isinstance(val, list):
+                        for elem in val:
+                            result.add(elem)
+                    else:
+                        result.add(val)
+
+                return result
+
+            all_fields = _to_set(y_values.values())
+            if (
+                all([isinstance(field, str) for field in all_fields])
+                and len(all_fields) == 1
+            ):
+                # if we use the same field from all files,
+                # we dont have to generate it
+                for datapoint in datapoints:
+                    result.append(
+                        _add_version_info(
+                            datapoint, _version_info(revision, filename)
+                        )
+                    )
+                properties_update.update({"y": all_fields.pop()})
+            else:
+                for def_filename, val in y_values.items():
+                    if isinstance(val, str):
+                        fields = [val]
+                    if isinstance(val, list):
+                        fields = val
+                    for datapoint in datapoints:
+                        for field in fields:
+                            if field in datapoint and def_filename in filename:
+                                result.append(_generate_y(datapoint, field))
+
+        self.inferred_properties = {
+            **self.inferred_properties,
+            **properties_update,
+        }
+
+        return result
```

Comment:
If I make a typo in a stage plot:

```yaml
stages:
  foo:
    cmd: python foo.py
    plots:
      - dvclive/scalars/foo.tsv:
          y: foos  # Should be foo
```

I get an explicit error `ERROR: unexpected error - Field 'foos' does not exist in provided data.`.

However, if I make a typo in the definition of the flexible plot, all the steps pass silently but `result`is an empty list and no error is raised:

```yaml
plots:
  foo:
    y: 
      dvclive/scalars/foo.tsv: foos  # Should be foo
```

Reply:
Investigating, 
Note: 
```yaml
plots:
  plots.csv:
    y: foos
```
seems to be working as intended

Reply:
@daavoo could you provide some info on your setup? I tried with non-dvclive setup, then, thinking that's some `dvclive` issue I did also with `dvclive` but in both cases I get expected error.

Reply:
My script:

```bash
#!/bin/bash

set -exu
pushd $TMPDIR

wsp=test_wspace
rep=test_repo

rm -rf $wsp && mkdir $wsp && pushd $wsp
main=$(pwd)

mkdir $rep && pushd $rep

git init 
dvc init

echo -e "step,foo
1,0.1
2,0.2
3,0.25
" > data.csv

echo -e "from dvclive import Live
l = Live('logs')
l.log('val', 1)
l.next_step()
l.log('val', 2)" > code.py


dvc add data.csv
git add -A
git commit -am "initial"

dvc run -n train --plots plots.csv -d data.csv cp data.csv plots.csv
dvc plots modify plots.csv -y foos

dvc run -n log --outs logs -d code.py python code.py

git add -A
git commit -am "second"

echo -e "plots:
  foo:
    y: 
      logs/scalars/val.tsv: foos" >> dvc.yaml

dvc plots show

```

Reply:
@pared please see my last comment. The issue is when using multiple `y`:

Reply:
I can confirm that this problem exists. I spend some time thinking about it and I am not sure we should make `dvc-render` raise this error. We had few changes aiming to do "as much as possible" rendering before (#5525). Now, if there is no particular column we will fail `show/diff` command even if other provided plots are ok. I think we might need to adjust renderers to log warnings and skipping rendering to keep up with this approach. 

What do you think @daavoo?

Reply:
Sounds good to me. Let's open a separate issue for it